patches/tModLoader/Terraria/ModLoader/ModPlayer.cs

```diff
@@ -927,5 +916,23 @@ public class ModPlayer
 		/// <param name="price">The price the player paid in copper coins.</param>
 		public virtual void PostNurseHeal(NPC nurse, int health, bool removeDebuffs, int price) {
 		}
+
+		/// <summary>
+		/// Called when the player is created in the menu.
+		/// You can use this method to add items to the player's starting inventory, as well as their inventory when they respawn in mediumcore.
+		/// </summary>
+		/// <param name="mediumCoreDeath">Whether you are setting up a mediumcore player's inventory after their death.</param>
+		/// <returns>An enumerable of the items you want to add. If you want to add nothing, return an empty enumerable.</returns>
+		public virtual IEnumerable<Item> AddStartingItems(bool mediumCoreDeath) {
+			return Enumerable.Empty<Item>();
+		}
+
+		/// <summary>
+		/// Allows you to modify the items that will be added to the player's inventory. Useful if you want to stop vanilla or other mods from adding an item.
```

Comment:
Maybe you can add: You can access a mod's items by using the mod's internal name as the indexer: `additions["ModName"]`. To access vanilla items you can use "Terraria" as the indexer.